### PR TITLE
Support VLM in chat completion (+some specs updates)

### PR DESCRIPTION
**Pull request by Wauplin**
Follow-up PR after https://github.com/huggingface/huggingface.js/pull/915.

Updates chat completion input parameters (tools and `stream_options`) + add supports for VLMs => sending image as text message.

Also adds `TextToSpeech` spec definition but not used at the moment (same specs as `TextToAudio`).

Also added an example in docs:
```py
from huggingface_hub import InferenceClient

# works with remote url or base64 encoded url
image_url ="https://cdn.britannica.com/61/93061-050-99147DCE/Statue-of-Liberty-Island-New-York-Bay.jpg"

client = InferenceClient("mistralai/Mistral-Nemo-Instruct-2407")
output = client.chat.completions.create(
    messages=[
        {
            "role": "user",
            "content": [
                {
                    "type": "image_url",
                    "image_url": {"url": image_url},
                },
                {
                    "type": "text",
                    "text": "Describe this image in one sentence.",
                },
            ],
        },
    ],
)

print(output.choices[0].message.content)
#A determine figure of Lady Liberty stands tall, holding a torch aloft, atop a pedestal on an island.
```

### CODE REVIEW

**Comment by MoritzLaurer** [comment]
Cool! One great addition would be to also explicitly document how people can use a local, non-public image in PIL and convert it to a base64 string which the API also understands. People often get confused about this. I use code similar to this: 

```py
import base64
from io import BytesIO
from PIL import Image

# Open an image using PIL
image = Image.open('path_to_your_image.png')

def image_to_data_url(image):
    # Use the image's format or default to 'PNG' if format is None
    format = image.format or 'PNG'
    with BytesIO() as buffer:
        image.save(buffer, format=format)
        img_str = base64.b64encode(buffer.getvalue()).decode('utf-8')
    return f"data:image/{format.lower()};base64,{img_str}"

messages = [
    {
        "role": "user",
        "content": [
            {
                "type": "text",
                "text": "some prompt"
            },
            {
                "type": "image_url",
                "image_url": {
                    "url": image_to_data_url(image)
                },
            }
        ]
    }
]

```

**Comment by Wauplin** [comment]
I'll see how I can integrate that in the docs :+1: I don't want the snippet to be bloated with 3 different use cases. For now I have:

```py
>>> image_path = "/path/to/image.jpeg"
>>> with open(image_path, "rb") as f:
...     base64_image = base64.b64encode(f.read()).decode("utf-8")
>>> image_url = f"data:image/jpeg;base64,{base64_image}"
```

I'm thinking that in addition to the method docstring we should maybe add a section in the [Inference guide](https://huggingface.co/docs/huggingface_hub/guides/inference) typically to explain better how to deal with system prompt, tools, images, etc.

**Review by hanouticelina — APPROVED** [review]
All good for me! thanks @Wauplin

**Review by LysandreJik — APPROVED** [review]
Thank you @Wauplin

File: src/huggingface_hub/inference/_generated/_async_client.py [hunk 1]
```diff
@@ -734,6 +741,41 @@ async def chat_completion(
...
+
+        # provide a remote URL
+        >>> image_url ="https://cdn.britannica.com/61/93061-050-99147DCE/Statue-of-Liberty-Island-New-York-Bay.jpg"
+        # or a base64-encoded image
+        >>> image_path = "/path/to/image.jpeg"
+        >>> with open(image_path, "rb") as f:
+        ...     base64_image = base64.b64encode(f.read()).decode("utf-8")
+        >>> image_url = f"data:image/jpeg;base64,{base64_image}"
+
+        >>> client = AsyncInferenceClient("mistralai/Mistral-Nemo-Instruct-2407")
```
Comment (osanseviero) [inline src/huggingface_hub/inference/_generated/_async_client.py:757]: should this be pixtral? nemo instruct is text-only

**Review by osanseviero — APPROVED** [review]
Nice!

File: src/huggingface_hub/inference/_generated/types/audio_classification.py [hunk 2]
```diff
@@ -27,8 +27,10 @@ class AudioClassificationParameters(BaseInferenceType):
 class AudioClassificationInput(BaseInferenceType):
     """Inputs for Audio Classification inference"""
 
-    inputs: Any
-    """The input audio data"""
+    inputs: str
+    """The input audio data as a base64-encoded string. If no `parameters` are provided, you can
```
Comment (hanouticelina) [inline src/huggingface_hub/inference/_generated/types/audio_classification.py:31]: I just realized that `client.audio_classification()` current implementation can take also a local audio file, or a URL pointing to an audio file as input (thanks to the [_open_as_binary()](https://github.com/huggingface/huggingface_hub/blob/c0fd4e0f7519a4e3659c836081cc7e38c0d14b35/src/huggingface_hub/inference/_client.py#L288) context manager). We also handle base64 encoding in the code. This is actually the case for other tasks that take image or audio as input.

While it's quite convenient for users to not worry about file handling or encoding, it makes the documentation of the library not aligned with the current API specs for the `inputs` field.

File: src/huggingface_hub/inference/_generated/_async_client.py (hunk 1 above)
Comment (Wauplin) [reply src/huggingface_hub/inference/_generated/_async_client.py:757]: Definitely an oversight from me. It seems that when the model is an LLM (not VLM), the data is still transferred to as text. In my case the URL contains "statue of liberty" so the LLM has been able to describe it (and trick me into thinking the image has been used).

Comment (osanseviero) [reply src/huggingface_hub/inference/_generated/_async_client.py:757]: LLM: 1
Humans: 0

File: src/huggingface_hub/inference/_generated/types/audio_classification.py (hunk 2 above)
Comment (hanouticelina) [reply src/huggingface_hub/inference/_generated/types/audio_classification.py:31]: @Wauplin as discussed, this will be addressed in #2561
LGTM :)

**Comment by Wauplin** [comment]
Thanks for the reviews! I've updated the example to use `"HuggingFaceM4/idefics2-8b-chatty"` for now.

**Pull request merged**